rotkehlchen/chain/polygon_pos/modules/polygon_pos_bridge/decoder.py

```diff
@@ -0,0 +1,182 @@
+                asset=asset,
+                balance=Balance(amount=amount),
+                location_label=user_address,
+                notes=f'Bridge {amount} {asset.resolve_to_asset_with_symbol().symbol} from Ethereum to Polygon POS via Polygon bridge',  # noqa: E501
+                counterparty=CPT_POLYGON,
+                address=context.tx_log.address,
+            ))
+
+        return DEFAULT_DECODING_OUTPUT
+
+    def _decode_deposit(self, context: DecoderContext) -> DecodingOutput:
+        """Decodes bridge deposit events.
+        The transfer event has not been decoded yet and must be transformed via an action item.
+
+        This is triggered by WITHDRAW_FOUR_BYTES and ERC20_OR_ERC721_TRANSFER, which is quite open.
+        To narrow it down we check for the following:
+            - transaction has two logs
+            - first log is Transfer, second is LogFeeTransfer emitted by POL
+            - transfer's from address is a tracked address and to address is ZERO_ADDRESS
+        Unfortunately there's nothing else special in the logs to filter against.
+        """
+        if (
+            len(context.all_logs) == 2 and
+            context.all_logs[0].topics[0] == ERC20_OR_ERC721_TRANSFER and
+            context.all_logs[1].topics[0] == LOG_FEE_TRANSFER_TOPIC and
+            context.all_logs[1].address == POL_TOKEN_ADDRESS and
+            self.base.is_tracked(user_address := bytes_to_address(context.tx_log.topics[1])) and
+            bytes_to_address(context.tx_log.topics[2]) == ZERO_ADDRESS
+        ):
+            asset = self.base.get_or_create_evm_asset(context.tx_log.address)
+            amount = asset_normalized_value(
+                amount=int.from_bytes(context.tx_log.data[0:32]),
+                asset=asset,
+            )
+            context.action_items.append(ActionItem(
+                action='transform',
+                from_event_type=HistoryEventType.SPEND,
+                from_event_subtype=HistoryEventSubType.NONE,
+                asset=asset,
+                amount=amount,
+                location_label=user_address,
+                to_event_type=HistoryEventType.DEPOSIT,
+                to_event_subtype=HistoryEventSubType.BRIDGE,
+                to_notes=f'Bridge {amount} {asset.resolve_to_asset_with_symbol().symbol} from Polygon POS to Ethereum via Polygon bridge',  # noqa: E501
+                to_counterparty=CPT_POLYGON,
+            ))
+
+        return DEFAULT_DECODING_OUTPUT
+
+    def _decode_plasma_deposit(self, context: DecoderContext) -> DecodingOutput:
+        """Decodes deposits to plasma bridge."""
+        for event in context.decoded_events:
+            if (
+                event.event_type == HistoryEventType.SPEND and
+                event.event_subtype == HistoryEventSubType.NONE and
+                event.location_label is not None and
+                self.base.is_tracked(user_address := string_to_evm_address(event.location_label)) and  # noqa: E501
```

Comment:
same comment for location_label and is_tracked as the other methods. (notice that I did not say this comment in `_decode_deposit` as that does not match already created logs, so there we can't be sure and the check there is probably good)